tests/test_market.py

```diff
@@ -0,0 +1,423 @@
+            wallet_address=1,
+            trade_amount=agent.amount_to_trade,
+        )
+        actual_deltas = Deltas(market_deltas=market_deltas, agent_deltas=agent_deltas)
+        self.compare_deltas(actual_deltas=actual_deltas, expected_deltas=expected_deltas)
+
+    def run_market_test_open_short(self, agent_policy, expected_deltas: Deltas):
+        """Test market trades result in the expected wallet balances"""
+        simulator = self.set_up_test(agent_policies=[agent_policy])
+        agent = simulator.agents[1]
+        market_deltas, agent_deltas = simulator.market.open_short(
+            wallet_address=1,
+            trade_amount=agent.amount_to_trade,
+        )
+        actual_deltas = Deltas(market_deltas=market_deltas, agent_deltas=agent_deltas)
+        self.compare_deltas(actual_deltas=actual_deltas, expected_deltas=expected_deltas)
+
+    def run_market_test_close_long(self, agent_policy, expected_deltas: Deltas):
+        """Test market trades result in the expected wallet balances"""
+        simulator = self.set_up_test(agent_policies=[agent_policy])
+        agent = simulator.agents[1]
+        market_deltas, agent_deltas = simulator.market.open_long(
+            wallet_address=1,
+            trade_amount=agent.amount_to_trade,  # in base: that's the thing in your wallet you want to sell
+        )
+        # peek inside the agent's wallet and see how many bonds they have
+        amount_of_bonds_purchased = agent_deltas.longs[0].balance
+        # sell those bonds to close the long
+        market_deltas, agent_deltas = simulator.market.close_long(
+            mint_time=0,
+            wallet_address=1,
+            trade_amount=amount_of_bonds_purchased,  # in bonds: that's the thing in your wallet you want to sell
+        )
+        actual_deltas = Deltas(market_deltas=market_deltas, agent_deltas=agent_deltas)
+        self.compare_deltas(actual_deltas=actual_deltas, expected_deltas=expected_deltas)
+
+    def run_market_test_close_short(
+        self, agent_policy, expected_deltas: Deltas, delete_logs=True, partial=None, tick_time=None
+    ):  # pylint: disable=too-many-arguments
+        # disabling pylint because this function is used in 3 tests w/ tiny parameter tweaks
+        """Test market trades result in the expected wallet balances"""
+        simulator = self.set_up_test(agent_policies=[agent_policy])
+        agent = simulator.agents[1]
+        market_deltas, agent_deltas = simulator.market.open_short(
+            wallet_address=1,
+            trade_amount=agent.amount_to_trade,  # in bonds: that's the thing you want to short
+        )
+        # peek inside the agent's wallet and see how many bonds they have
+        amount_of_bonds_sold = agent_deltas.shorts[0].balance
+        # sell pthose bonds to close the short (partial amount if selected)
+        trade_amount = amount_of_bonds_sold * partial if partial else amount_of_bonds_sold
```

Comment:
I think it would be better to just set the default of `partial` to `1` and then you don't need the conditional.

Beyond that, checks for `None` should be done as `if partial is not None`. See here:
https://peps.python.org/pep-0008/#programming-recommendations

Reply:
setting parameter to false avoids this issue, i believe